app/models/xml/manifest.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module XML
+  class Manifest
+    attr_reader :document
+
+    def initialize(translation)
+      @translation = translation
+
+      @document = Nokogiri::XML(@translation.resource.manifest)
+      @document.root = create_manifest if @document.root.nil?
+
+      manifest_node = @document.root
+      add_manifest_metadata(manifest_node)
+      insert_translated_name(manifest_node)
+    end
+
+    def add_page(filename, sha_filename)
+      pages_node.add_child(create_file_node(XmlUtil::XMLNS_MANIFEST, 'page', filename, sha_filename))
+    end
+
+    def add_resource(filename, sha_filename)
+      resources_node.add_child(create_file_node(XmlUtil::XMLNS_MANIFEST, 'resource', filename, sha_filename))
+    end
+
+    private
+
+    def create_manifest
+      @document.root = @document.create_element('manifest', xmlns: XmlUtil::XMLNS_MANIFEST)
+    end
+
+    def add_manifest_metadata(manifest_node)
+      manifest_node['tool'] = @translation.resource.abbreviation
+      manifest_node['locale'] = @translation.language.code
+      manifest_node['type'] = @translation.resource.resource_type.name
+    end
+
+    def insert_translated_name(manifest_node)
+      title_node = XmlUtil.xpath_namespace(manifest_node, 'manifest:title').first
+      return if title_node.nil?
+
+      name_node = title_node.xpath('content:text[@i18n-id]').first
+      name_node.content = @translation.translated_name
+    end
+
+    def pages_node
+      return @pages if @pages
+      @pages = XmlUtil.get_or_create_child(@document.root, XmlUtil::XMLNS_MANIFEST, 'pages')
```

Comment:
Won't this always `create?`  Why do you need the `get` part?

Reply:
we actually ran into an issue where multiple pages nodes existed in the manifest since they were in the raw manifest xml.

Reply:
Hmm... can you elaborate a little?

Reply:
the manifest for the four was entered as:
```
<manifest>
  <title />
  <pages />
</manifest>
```
when the mobile-content-api processed this manifest it created a second `<pages>` node. This broke the web renderer and also isn't valid with the spec.

In addition one of the changes coming for articles is including an `<article:aem-import src="">` node within the `<pages>` node, we haven't fully figured out how we will manage that yet, but initially this will probably be defined in the manifest xml stored in the database.

Reply:
Ohhh I get it.  Cool!